Python/callback.hh

```diff
@@ -0,0 +1,76 @@
+/*-
+ * Copyright (c) 2016 Alfredo Mazzinghi
+ *
+ * All rights reserved.
+ *
+ * This software was developed by SRI International and the University of
+ * Cambridge Computer Laboratory under DARPA/AFRL contract FA8750-10-C-0237
+ * ("CTSRD"), as part of the DARPA CRASH research programme.
+ *
+ * This software was developed by SRI International and the University of
+ * Cambridge Computer Laboratory under DARPA/AFRL contract FA8750-11-C-0249
+ * ("MRC2"), as part of the DARPA MRC research programme.
+ *
+ * @BERI_LICENSE_HEADER_START@
+ *
+ * Licensed to BERI Open Systems C.I.C. (BERI) under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  BERI licenses this
+ * file to you under the BERI Hardware-Software License, Version 1.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *   http://www.beri-open-systems.org/legal/license-1-0.txt
+ *
+ * Unless required by applicable law or agreed to in writing, Work distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * @BERI_LICENSE_HEADER_END@
+ */
+
+#include "../streamtrace.hh"
+
+namespace cheri
+{
+namespace streamtrace
+{
+
+/**
+ * Proxy classes to dispatch callbacks from cheritrace and
+ * the SWIG target languages (see SWIG directors).
+ * The c_scanner method is bound to the director object and
+ * given to the trace::scan method, the c_scanner in turns invokes
+ * Scanner::run which should be implemented by the callback class
+ * in the target language.
+ *
+ * XXX: The three proxy classes may be merged in a template
+ */
+
+struct Scanner
```

Comment:
I'm not sure why you need these.  You should be able to use any class that defines `operator::()` (including a lambda) and you ought to be able to do this in a private namespace for the bindings, rather than defining a new class.

Looking at your bindings code, can't you just pass a lambda to the scan functions that calls back into the Python interpreter with the Python function that you pass?